mountpoint-s3/src/data_cache/disk_data_cache.rs

```diff
@@ -26,13 +26,30 @@ const HASHED_DIR_SPLIT_INDEX: usize = 2;
 
 /// On-disk implementation of [DataCache].
 pub struct DiskDataCache {
-    block_size: u64,
     cache_directory: PathBuf,
-    limit: CacheLimit,
+    config: DiskDataCacheConfig,
     /// Tracks blocks usage. `None` when no cache limit was set.
     usage: Option<Mutex<UsageInfo<DiskBlockKey>>>,
 }
 
+/// Configuration for a [DiskDataCache].
+#[derive(Debug)]
+pub struct DiskDataCacheConfig {
+    /// Size of data blocks.
+    pub block_size: u64,
+    /// How to limit the cache size.
+    pub limit: CacheLimit,
+}
+
+impl Default for DiskDataCacheConfig {
+    fn default() -> Self {
+        Self {
+            block_size: 1024 * 1024,                               // 1 MiB block size
+            limit: CacheLimit::AvailableSpace { min_ratio: 0.01 }, // Preserve 1% available space
```

Comment:
default should be larger because of FS metadata overhead (inodes etc), let's do 5%